pyanaconda/modules/common/task/__init__.py

```diff
@@ -38,15 +38,22 @@ def publish_task(message_bus, namespace, task_instance):
     return object_path
 
 
-def sync_run_task(task_proxy):
+def sync_run_task(task_proxy, callback=None):
     """Run a remote task synchronously.
 
+    The given callback will be called every iteration.
+
     :param task_proxy: a proxy of the remote task
+    :param callback: a callback
     :raise: a remote error
     """
     task_proxy.Start()
 
     while task_proxy.IsRunning:
+
+        if callback:
+            callback(task_proxy)
+
```

Comment:
What processing do you mean? This is a synchronous call so I think it should be just blocked until task is finished. When you want to react on that than the asynchronous processing should be used instead. This way you basically doing asynchronous call from synchronous call.

Do you have some use-case for this because I can't think of any?

Reply:
This functionality is used here: https://github.com/rhinstaller/anaconda/pull/1507/files#diff-dc67ecb027c0c13b20bd1bd133953322R146

The callback is used to show progress of a task that is running synchronously. However, there might be a better solution.

Reply:
I didn't find a better solution.